oioioi/contests/utils.py

```diff
@@ -250,6 +260,87 @@ def visible_rounds(request, no_admin=False):
         request, r, no_admin=no_admin,
     )]
 
+@make_request_condition
+@request_cached
+def are_rules_visible(request):
+    return (
+        hasattr(request, 'contest')
+        and request.contest.show_contest_rules
+    )
+
+@request_cached
+def get_number_of_rounds(request):
+    """Returns the number of rounds in the current contest.
+    """ 
+    return Round.objects.filter(contest=request.contest).count()
+
+
+def get_contest_dates(request):
+    """Returns the end_date of the latest round and the start_date
+    of the earliest round.
+    """ 
+    rtimes = rounds_times(request, request.contest)
+
+    ends = [
+        rt.get_end()
+        for rt in rtimes.values()
+    ]
+    starts = [
+        rt.get_start()
+        for rt in rtimes.values()
+    ]
+
+    if starts and None not in starts:  
+        min_start = min(starts)  
+    else:  
+        min_start = -1
+
+    if ends and None not in ends:  
+        max_end = max(ends)  
+    else:  
+        max_end = -1  
+        
+
+    return min_start, max_end
+
+
+@request_cached
+def get_problems_sumbmission_limit(request):
```

Comment:
I encourage to move all logic here and leave just rendering in view / template.
Also I would recommend to reduce the information just to:
- no limit
- X submission per problem (if there is only single limit for all tasks)
- X to Y submissions depending on the problem (if there are multiple different values)